src/libraries/System.Private.CoreLib/src/System/Globalization/Normalization.cs

```diff
@@ -35,5 +56,63 @@ internal static string Normalize(string strInput, NormalizationForm normalizatio
                 NlsNormalize(strInput, normalizationForm) :
                 IcuNormalize(strInput, normalizationForm);
         }
+
+        internal static bool TryNormalize(ReadOnlySpan<char> source, Span<char> destination, out int charsWritten, NormalizationForm normalizationForm = NormalizationForm.FormC)
+        {
+            CheckNormalizationForm(normalizationForm);
+
+            if (GlobalizationMode.Invariant || source.IsEmpty)
+            {
+                // In Invariant mode we assume all characters are normalized.
+                // This is because we don't support any linguistic operation on the strings
+                charsWritten = 0;
+                return true;
```

Comment:
If we assume all characters are normalized, shouldn't we be copying them to the destination and setting charsWritten to the number of characters copied? Seeing charsWritten to 0 and returning true suggests the input normalized to empty. How is a dev supposed to consume that?